### PR TITLE
[NetStandard Unification] Username / Password flow

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: msal/src/MSAL.Common.props [hunk 1]
```diff
@@ -182,6 +180,7 @@
     <PackageReference Include="System.Runtime.Serialization.Json" Version="4.3.0" />
     <PackageReference Include="System.Xml.XDocument" Version="4.3.0" />
     <PackageReference Include="System.Net.NameResolution" Version="4.3.0" />
+    <PackageReference Include="System.Security.SecureString" Version="4.3.0" />
```
Comment (bgavrilMS) [inline msal/src/MSAL.Common.props:183]: Unfortunately we have to take a dependency on SecureString in order to throw nice exceptions on Android, UWP and iOS...

File: msal/src/Microsoft.Identity.Client/Features/UsernamePassword/IPublicClientApplication.cs [hunk 2]
```diff
@@ -25,17 +25,13 @@
 //
 //------------------------------------------------------------------------------
 
-using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Microsoft.Identity.Client
 {
-    /// <summary>
```
Comment (bgavrilMS) [inline msal/src/Microsoft.Identity.Client/Features/UsernamePassword/IPublicClientApplication.cs:34]: Main interface (this is a partial) comments are nicer :)

File: msal/src/Microsoft.Identity.Client/Features/UsernamePassword/PublicClientApplication.cs [hunk 3]
```diff
@@ -33,11 +33,15 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Identity.Core.WsTrust;
+using System;
 
 namespace Microsoft.Identity.Client
 {
+#if !ANDROID_BUILDTIME && !iOS_BUILDTIME && !WINDOWS_APP_BUILDTIME
```
Comment (MarkZuber) [inline msal/src/Microsoft.Identity.Client/Features/UsernamePassword/PublicClientApplication.cs:40]: This looks good for now but I'd really like to look towards a path for feature flags and having the particular platforms for a feature defined in the props file

**Review by MarkZuber — APPROVED** [review]

File: msal/src/MSAL.Common.props [hunk 4]
```diff
@@ -74,7 +74,7 @@
     <DefineConstants>$(DefineConstants);HAVE_ASYNC;HAVE_BINARY_SERIALIZATION;HAVE_COVARIANT_GENERICS;HAVE_DATA_CONTRACTS;HAVE_DATE_TIME_OFFSET;HAVE_DYNAMIC;HAVE_EXPRESSIONS;HAVE_FSHARP_TYPES;HAVE_GUID_TRY_PARSE;HAVE_HASH_SET;HAVE_IGNORE_DATA_MEMBER_ATTRIBUTE;HAVE_INOTIFY_COLLECTION_CHANGED;HAVE_INOTIFY_PROPERTY_CHANGING;HAVE_ISET;HAVE_LINQ;HAVE_METHOD_IMPL_ATTRIBUTE;HAVE_NON_SERIALIZED_ATTRIBUTE;HAVE_READ_ONLY_COLLECTIONS;HAVE_REFLECTION_BINDER;HAVE_SERIALIZATION_BINDER_BIND_TO_NAME;HAVE_STRING_JOIN_WITH_ENUMERABLE;HAVE_TIME_SPAN_PARSE_WITH_CULTURE;HAVE_TIME_SPAN_TO_STRING_WITH_CULTURE;HAVE_TIME_ZONE_INFO;HAVE_UNICODE_SURROGATE_DETECTION;HAVE_VARIANT_TYPE_PARAMETERS;HAVE_VERSION_TRY_PARSE;HAVE_XLINQ;HAVE_XML_DOCUMENT;HAVE_CONCURRENT_DICTIONARY;HAVE_ICONVERTIBLE;$(AdditionalConstants)</DefineConstants>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetFramework)' == 'netstandard1.3'">
-    <DefineConstants>$(DefineConstants);NETSTANDARD</DefineConstants>
+    <DefineConstants>$(DefineConstants);R</DefineConstants>
```
Comment (MarkZuber) [inline msal/src/MSAL.Common.props:77]: i'm guessing this isn't intentional that you're defining R as a constant?

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by bgavrilMS** [comment]
Will rebase and merge to dev manually

**Pull request closed without merging**